### ISSUE TITLE
✍️ Contribution period: Chiamaka Ohaji 

### ISSUE THREAD

**Issue by Chiamakaj** [ersilia-os/ersilia#983]
### Week 1 - Get to know the community

- [X] Join the communication channels
- [X] Open a GitHub issue (this one!)
- [X] Install the Ersilia Model Hub and test the simplest model
- [x] Install Docker if needed, and test another model
- [x] Write a motivation statement to work at Ersilia
- [x] Submit your first contribution to the Outreachy site

### Week 2 - Get Familiar with Machine Learning for Chemistry

- [x] Select a model from the list suggested in GitBook
- [x] Download and serve the model via the Ersilia Model Hub to ensure it works
- [x] Open a repository on your GitHub user with all the necessary files
- [x] Select and clean a dataset of 1000 molecules (example notebook 1)
- [x] Run predictions for the molecules on the selected model and evaluate the results

### Week 3 - Validate a Model in the Wild

- [ ] Find a suitable dataset with sufficient experimental results
- [ ] Clean and standardize the dataset
- [ ] Run predictions and calculate metrics.

### Week 4 - Prepare your final application

- [ ] Submit the final application in the Outreachy website

**Comment by Chiamakaj** [comment]
I decided to join Outreachy after I read on their website introductory page that “Outreachy provides internships to people subject to systemic bias and impacted by underrepresentation in the technical industry where they are living”. This added to the courage I needed to push forward in my career pursuits. Also, contributing to open-source projects strengthens the basic idea of sharing knowledge and resources for the betterment of all. Working on open-source projects begets fulfillment as a human adding value to other humans.
I am currently a master's student studying Information Technology with a concentration in Machine learning. Artificial Intelligence. I am currently doing a machine learning course where we are not only taught the mathematics behind models but also build models from scratch. Seeing Ersilia’s project immediately caught my attention because this will be a beautiful blend of my knowledge and career aspirations. I have worked on several projects where I used pandas, numpy, Pytorch, OpenStack, Azure, C++, and shell scripting. I believe that these projects prepared me for this opportunity with Ersilia.
My passion for health sciences, combined with the potential for ground-breaking advancements through the integration of ML/AI, motivates my interest in Ersilia's initiative. I believe that this will play a crucial role in addressing critical health concerns and fostering positive changes in our rapidly evolving environment.
Ersilia's project is an exciting opportunity for me to apply my current skills and knowledge in a meaningful way. Working on this project aligns perfectly with my long-standing goal of making tangible contributions to human health.
Interning with Ersilia is not just an opportunity for me to pursue my passion but also to advance my career. The hands-on experience gained through this internship will complement my academic learning, providing me with real-world insights as to where technology and health meet. During the internship, I will collaborate with the Ersilia team on innovative solutions and learn from experienced professionals in the field. 
After the internship, I hope to continue contributing to open-source projects in technology and health sciences. With the experience gained at Ersilia, I will pursue a career that allows me to drive positive change, leveraging artificial intelligence to address pressing health challenges.

**Comment by Ajoke23** [comment]
Well-done @Chiamakaj. This is really amazing

**Comment by Chiamakaj** [comment]
> Well-done @Chiamakaj. This is really amazing
Thanks Ajoke

**Comment by DhanshreeA** [comment]
Hi @Chiamakaj amazing work so far! I think you can link this in your final application and start working on it. I have a few comments though, please address them:
1. Could you add a README within the data folder mentioning the exact steps you took to obtain the external dataset for model validation task?
2. Could you also comment on the number of smiles that were common within the training dataset and the external validation dataset?
3. Btw, nice use of the PrettyTable module! :)

**Comment by Chiamakaj** [comment]
> Hi @Chiamakaj amazing work so far! I have a few comments, please address them:
> 
> 1. Could you add a README within the data folder mentioning the exact steps you took to obtain the external dataset for model validation task?
> 2. Could you also comment on the number of smiles that were common within the training dataset and the external validation dataset?
> 3. I don't see an implementation of the model from the publication, rather only the use of the ersilia model hub implementation. Any reason why that's the case? I might have missed it during the review, so please point me to the code if that's the case.
> 4. Btw, nice use of the PrettyTable module! :)

@DhanshreeA Thanks for the review and feedback. 
I have added a README file within the data folder as instructed and commented on the number of smiles that were common during the model validation using the external dataset.
I have added the implementation file for the original model implementation to the repository [here](https://github.com/Chiamakaj/Ersilia_model_validation/blob/main/notebooks/cardiotox_implementation.py) and added comments on this to the original README file. The output from the original model implementation was used to carry out the metric reproducibility in the task but I forgot to add the Python file used during this process.

Kindly confirm and share any other feedback.

**Comment by DhanshreeA** [comment]
Hi @Chiamakaj, great work so far and your points above make sense! I only have one small feedback: while filtering common smiles between the external dataset and the training dataset, I would recommend doing that using InchiKeys and not SMILES as that is more robust. After that, you can work on your final application!

**Comment by Chiamakaj** [comment]
> Hi @Chiamakaj, great work so far and your points above make sense! I only have one small feedback: while filtering common smiles between the external dataset and the training dataset, I would recommend doing that using InchiKeys and not SMILES as that is more robust. After that, you can work on your final application!

@DhanshreeA , as instructed, I have included your recommendation and looked for common molecules based on InchiKeys. Thank you.

**Comment by Chiamakaj** [comment]
@DhanshreeA Kindly review and give feedback on my final application to Outreachy in the link [here](https://docs.google.com/document/d/1uFHSXg66NWkN_1n78WjLYJWdrqIpiQcP-dQTupQZmJ0/edit). Thank you.

**Comment by Chiamakaj** [comment]
# Outreachy Final Application (review)

@DhanshreeA Kindly review and give feedback on my final application to Outreachy in the link [here](https://docs.google.com/document/d/1uFHSXg66NWkN_1n78WjLYJWdrqIpiQcP-dQTupQZmJ0/edit). Thank you.
